src/algorithms/core.ts

```diff
@@ -116,24 +119,41 @@ export interface ViewportAlgorithmOptions extends AlgorithmOptions {
  */
 export abstract class AbstractViewportAlgorithm extends AbstractAlgorithm {
   protected viewportPadding = 60;
+  protected state: { zoom: number };
```

Comment:
I guess this should be `zoom | null` (see l 127 below).

Notes:
- I have a pending PR changing the base code,
- we should probably investigate why this type error is not reported!